infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -343,6 +376,65 @@ public void prefixTrim(long address) {
         logMetadata.prefixTrim(address);
     }
 
+    @Override
+    public void suffixTrim() {
+
+        // Have a global tail copy is necessary, since degrade to reset is dependent on it
+        long globalTailCopy = logMetadata.getGlobalTail();
+
+        try {
+            long committedTail = dataStore.getCommittedTail();
+
+            // Check if committed tail is not less than starting address
+            if (isTrimmed(committedTail)) {
+                throw new SuffixTrimException("local committed tail is trimmed",
+                        committedTail, globalTailCopy);
+            }
+
+            // Check if committed tail is greater than global tail
+            if (committedTail > globalTailCopy) {
+                throw new SuffixTrimException("local committed tail is greater than global tail",
```

Comment:
I think the message should start with a capital letter.

Reply:
Also, how is this exception being handled? Looks like the user will get CorfuMsgType.ERROR_SERVER_EXCEPTION and the log unit server will throw a LogUnitException. Is this what you want? Maybe we could add a handler to the LogUnitHandler and give a client a more specific exception?

Reply:
This Exception only carries error message and data, and all runtime exceptions will be caught by below line 427. Once it gets an RuntimeException, it will degrade to reset. I am not sure if this is too conservative.

For some cases like committedTail > globalTail, maybe we can assume that all data is committed and keep suffixTrim. What do you think?